pkg/boot/gsurv/option.go

```diff
@@ -0,0 +1,72 @@
+package gsurv
+
+import (
+	"net"
+
+	"github.com/libp2p/go-libp2p-core/discovery"
+)
+
+type DialFunc func(net.Addr) (net.PacketConn, error)
+type ListenFunc func(net.Addr) (net.PacketConn, error)
+
+// Config supplies options to the dependency-injection framework.
+type Config struct {
```

Comment:
I suggest renaming this to `Transport`, since it's a more precise descriptor.  Moreover, this lets us collapse the options into a single `WithTransport`.

We might also consider changing the constructor's type signature to `New(h host.Host, t Transport, addr net.Addr) (GSurv, error)`, since we only have one option.  Users could pass in a zero-value `Transport{}`, which could fall back on a sensible default if `Dial` or `Listen` are nil.

Reply:
The reason for all of this is that I generally don't like to use functional options unless there are multiple options, which isn't currently the case.